### DISCUSSION TITLE
How can I use the `MaybeSend` type?

### DISCUSSION THREAD

**Discussion by shanebishop** [mlua-rs/mlua#429] [Q&A]
I want to wrap calls to [`mlua::Lua::create_function`](https://docs.rs/mlua/latest/mlua/struct.Lua.html#method.create_function) so that I can have generic error handling for creation of different functions.

Here is what I tried:
```rust
fn create_function<'lua, A, R, F>(lua: &'lua Lua, name: &str, func: F) -> bool
where
    A: FromLuaMulti<'lua>,
    R: IntoLuaMulti<'lua>,
    F: Fn(&'lua Lua, A) -> Result<R> + MaybeSend + 'static,
{
    let fun = lua.create_function(func);

    if let Ok(fun) = fun {
        lua.globals().set(name, fun);
        true
    } else {
        error!("Error creating function: {}", name);
        false
    }
}
```

However, I get this error:
```
error[E0405]: cannot find trait `MaybeSend` in this scope
   --> src/lib.rs:276:40
    |
276 |     F: Fn(&'lua Lua, A) -> Result<R> + MaybeSend + 'static,
    |                                        ^^^^^^^^^ not found in this scope
```

When I try adding `use mlua::types::MaybeSend;`, I get this error:
```
error[E0603]: module `types` is private
  --> src/lib.rs:19:11
   |
19 | use mlua::types::MaybeSend;
   |           ^^^^^  --------- trait `MaybeSend` is not publicly re-exported
   |           |
   |           private module
```

Is it possible for me to use the `MaybeSend` type? Or do I have no choice but to duplicate my error handling code for every call I make to `mlua::Lua::create_function`?

**Comment by khvzak** [comment]
`MaybeSend ` trait depends on mlua `send` feature flag.
If it set then `MaybeSend` can be replaced with `Send`, otherwise just omitted.
